frontend/server/src/main/java/org/pytorch/serve/job/GRPCJob.java

```diff
@@ -101,25 +122,59 @@ public void response(
-                responseObserver.onNext(reply);
-                if (cmd == WorkerCommands.PREDICT
-                        || (cmd == WorkerCommands.STREAMPREDICT
-                                && responseHeaders
-                                        .get(RequestInput.TS_STREAM_NEXT)
-                                        .equals("false"))) {
-                    responseObserver.onCompleted();
-                    logQueueTime();
-                } else if (cmd == WorkerCommands.STREAMPREDICT2
-                        && (responseHeaders.get(RequestInput.TS_STREAM_NEXT) == null
-                                || responseHeaders
-                                        .get(RequestInput.TS_STREAM_NEXT)
-                                        .equals("false"))) {
-                    logQueueTime();
+                // condition for OIP grpc ModelInfer Call
+                if (ConfigManager.getInstance().isOpenInferenceProtocol()) {
+                    if (((ServerCallStreamObserver<ModelInferResponse>) modelInferResponseObserver)
+                            .isCancelled()) {
+                        logger.warn(
+                                "grpc client call already cancelled, not able to send this response for requestId: {}",
+                                getPayload().getRequestId());
+                        return;
+                    }
+        
+                    Gson gson = new Gson();
+                    ModelInferResponse.Builder responseBuilder = ModelInferResponse.newBuilder();
+                    String jsonResponse = output.toStringUtf8();
+                    JsonObject jsonObject = gson.fromJson(jsonResponse, JsonObject.class);
+        
+                    responseBuilder.setId(jsonObject.get("id").getAsString());
+                    responseBuilder.setModelName(jsonObject.get("model_name").getAsString());
+                    responseBuilder.setModelVersion(jsonObject.get("model_version").getAsString());
+                    JsonArray jsonOutputs = jsonObject.get("outputs").getAsJsonArray();
+        
+                    for (JsonElement element : jsonOutputs) {
+                        InferOutputTensor.Builder outputBuilder = InferOutputTensor.newBuilder();
+                        outputBuilder.setName(element.getAsJsonObject().get("name").getAsString());
+                        outputBuilder.setDatatype(element.getAsJsonObject().get("datatype").getAsString());
+                        JsonArray shapeArray = element.getAsJsonObject().get("shape").getAsJsonArray();
+                        shapeArray.forEach(shapeElement -> outputBuilder.addShape(shapeElement.getAsLong()));
+                        setOutputContents(element, outputBuilder);
+                        responseBuilder.addOutputs(outputBuilder);
+        
+                    }
+                    modelInferResponseObserver.onNext(responseBuilder.build());
+                    modelInferResponseObserver.onCompleted();
```

Comment:
This section will cause problem when client send TS prediction request when openInference is enabled. 
could you create a separate command to handle this section?